### DISCUSSION TITLE
Pixels never occupied by keypoints

### DISCUSSION THREAD

**Discussion by SylwestrakLab** [talmolab/sleap#2031] [Help!]
Our tracking is working well at the macro scale.  However, when I plotted one keypoint across the whole experiment (in this case, elevated plus maze), I noticed that when I zooming into an example frame with the inference points overlaid from the whole experiment, only every other row/column of pixels is occupied by keypoints.  Is this an artifact of downsampling?  We are not downsampling our images at any point, and I had thought the network was trained on a downsampled image for the coarse identification of the subject, but then used a full image to identify the keypoints.  Thanks in advance for your help!
=
<img width="784" alt="Screenshot 2024-11-22 at 10 01 27 AM" src="https://github.com/user-attachments/assets/095157cf-d397-477f-bdac-96e4317c16e9">

**Comment by gitttt-1234** [comment]
Hi @SylwestrakLab,

This is the effect of quantization error, which happens when co-ordinates can't be resolved below a certain resolution. You could try decreasing your output stride to reduce this effect. I'm linking previous discussions which explains this issue in detail: #552, #554, #2006, #1739
